src/lgar.cxx

```diff
@@ -550,7 +553,29 @@ extern void InitFromConfigFile(string config_file, struct model_state *state)
   }
 
 
-  if (!is_giuh_ordinates_set) {
+  if (is_giuh_ordinates_set) {
+    int factor = int(1.0/state->lgar_bmi_params.timestep_h);
+
+    state->lgar_bmi_params.num_giuh_ordinates = factor*num_giuh_ordinates_temp;
+    state->lgar_bmi_params.giuh_ordinates = new double[state->lgar_bmi_params.num_giuh_ordinates+1];
+    
+    for (int i=0; i<num_giuh_ordinates_temp; i++) {
+      for (int j=0; j<factor; j++) {
+	int index = j + i * factor + 1;
+	state->lgar_bmi_params.giuh_ordinates[index] = giuh_ordinates_temp[i+1]/double(factor);
+      }
+    }
+    
+    if (verbosity.compare("high") == 0 || true) {
+      for (int i=1; i<=state->lgar_bmi_params.num_giuh_ordinates; i++)
+	std::cerr<<"GIUH ordinates (scaled) : "<<state->lgar_bmi_params.giuh_ordinates[i]<<"\n";
+      
+      std::cerr<<"          *****         \n";
+    }
+    
+    giuh_ordinates_temp = NULL;
```

Comment:
This is a memory leak - the allocation is never `delete[]`'d

Use `std::vector<double>` instead

Reply:
If the code really needs the memory back right at this point, it should then call
```suggestion
    giuh_ordinates_temp.clear();
```